src/QirRuntime/test/Standalone/InputTypes/qir-driver.cpp

```diff
@@ -0,0 +1,145 @@
+            "--anint",
+            anInt,
+            "An integer");
+
+        anIntOpt->required();
+
+        // Option for a Q# Double type.
+        double_t aDouble = 0.0;
+        CLI::Option *aDoubleOpt = app.add_option(
+            "--adouble",
+            aDouble,
+            "A double");
+
+        aDoubleOpt->required();
+
+        // Option for a Q# Bool type.
+        bool aBool = false;
+        CLI::Option *aBoolOpt = app.add_option(
+            "--abool",
+            aBool,
+            "A bool");
+
+        aBoolOpt->required();
+
+        // TODO: Add option for Q# Pauli type.
+        // TODO: Add option for Q# Result type.
+        // TODO: Add option for Q# String type.
+
+        // Option for a Q# Array<Int> type.
+        std::vector<int64_t> anIntegerArray;
+        CLI::Option *anIntegerArrayOpt = app.add_option(
+            "--anintegerarray",
+            anIntegerArray,
+            "An integer array");
+
+        anIntegerArrayOpt->required();
+
+        // TODO: Add option for Q# Tuple<T> type.
+
+        // With all the options added, parse arguments from the command line.
+        CLI11_PARSE(app, argc, argv);
+
+        // Redirect the simulator output from std::cout if the --simulator-output option is present.
+        std::ostream* simulatorOutputStream = &std::cout;
+        std::filebuf simulatorOutputFileBuffer;
+        if (!simulatorOutputFileOpt->empty()){
+            std::cout << "Simulator Output File: " << simulatorOutputFile << std::endl;
+            simulatorOutputFileBuffer.open(simulatorOutputFile, std::ios::out);
+            std::ostream simulatorOutputFileStream(&simulatorOutputFileBuffer);
+            // TODO: Call into the QIR runtime API to redirect the output of the simulator.
+            simulatorOutputStream = &simulatorOutputFileStream;
+        }
+
+        // TODO: This is for testing purposes only.
+        (*simulatorOutputStream) << "SIMULATOR OUTPUT:\nSample Output\n";
+
+        // Redirect the Q# opertion output from std::cout if the --operation-output option is present.
+        std::ostream* operationOutputStream = &std::cout;
+        std::filebuf operationOutputFileBuffer;
+        if (!operationOutputFileOpt->empty()){
+            std::cout << "Operation Output File: " << operationOutputFile << std::endl;
+            operationOutputFileBuffer.open(operationOutputFile, std::ios::out);
+            std::ostream operationOutputFileStream(&operationOutputFileBuffer);
+            operationOutputStream = &operationOutputFileStream;
+        }
+
+        // TODO: Remove this after the Message Q# function is integrated into the QIR runtime.
+        (*simulatorOutputStream) << anInt << std::endl;
+        (*simulatorOutputStream) << aDouble << std::endl;
+        (*simulatorOutputStream) << aBool << std::endl;
+        for (int64_t n : anIntegerArray) {
+            (*simulatorOutputStream) << n << " ";
+        }
+
+        (*simulatorOutputStream) << std::endl;
```

Comment:
FYI: `Message()` has been pushed to `main` in #511 ([PR](https://github.com/microsoft/qsharp-runtime/pull/511)).

Reply:
Thanks! I added that to the most recent changes and it works great :)